src/autorun.ts

```diff
@@ -72,6 +78,14 @@ export const run = <T>(fn: Cb<T>): Observable<T> => new Observable(observer => {
         } finally {
             context.$ = prev$;
             context._ = prev_;
+            // Remove all unused deps
+            for (let dep of deps.entries()) {
+                if (dep[1].used) {
+                    continue;
+                }
+                dep[1].subscription?.unsubscribe();
+                deps.delete(dep[0]);
```

Comment:
**minor**: `for-of` should support destruction syntax, like:

```ts
for(let [key,dep] of deps.entries())
```
that has bit better readability (though idk if/how it increases footprint for es5)

Reply:
Ah nice, I didn't know! I like esthetics 😁